storage/uri_operation_impossible.go

```diff
@@ -0,0 +1,16 @@
+package storage
+
+// Declare conformance with Error interface
+var _ error = URIOperationImpossible
```

Comment:
I'm sure it would be OK to have all declared errors in a single error.go file :)

Reply:
I agree, I also don't think the separate types is very go idiomatic. https://golang.org/src/io/io.go has some exported errors that I have used before;
```go
var EOF = errors.New("EOF")
```

Reply:
Discussing idomatic errors in Go is tricky as they are mid-transition ;) We need to go with what was best in 1.12 I think as some of the newer things we can't opt in to without upgrading our version required.
That said I don't remember what best practice was 3 versions ago.